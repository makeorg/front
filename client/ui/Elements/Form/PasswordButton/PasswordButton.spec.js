import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { PasswordButton } from './index';

describe('PasswordButton', () => {
  it('must match the snapshot with defaultProps', () => {
    const component = renderer.create(<PasswordButton />);
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(<PasswordButton tabIndex="-1" />);
    const PositiveTabIndex = renderer.create(<PasswordButton tabIndex="0" />);
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when password is displayed vs hidden', () => {
    const ShowPassword = renderer.create(
      <PasswordButton passwordIsDisplayed />
    );
    const HidePassword = renderer.create(
      <PasswordButton passwordIsDisplayed={false} />
    );
    expect(snapshotDiff(ShowPassword, HidePassword)).toMatchSnapshot();
  });
});
