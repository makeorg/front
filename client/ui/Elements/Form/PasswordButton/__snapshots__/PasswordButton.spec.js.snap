// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordButton must match the snapshot with defaultProps 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  padding: 0;
  border: none;
  background: transparent;
  background-color: transparent;
}

<button
  aria-hidden={true}
  className="c0"
  tabIndex={0}
>
  <svg
    icon="eye"
    lib="@fortawesome/react-fontawesome"
    prefix="fas"
  />
</button>
`;

exports[`PasswordButton must return the diff between snapshot when password is displayed vs hidden 1`] = `
Snapshot Diff:
- First value
+ Second value

  <button
    aria-hidden={true}
-   className="sc-1wm8iti-1 ntcd9w-9 fbihzC"
+   className="sc-1wm8iti-1 fncopd"
    tabIndex={0}
  >
    <svg
-     icon="eye-slash"
+     icon="eye"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </button>
`;

exports[`PasswordButton must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  <button
    aria-hidden={true}
    className="sc-1wm8iti-1 fncopd"
-   tabIndex="-1"
+   tabIndex="0"
  >
    <svg
      icon="eye"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
`;
